### PR TITLE
ArgoCD template draft

### PR DESCRIPTION
**Pull request by kalexand-rh**

### CODE REVIEW

File: scalability_and_performance/argocd.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+[id="argocd-what"]
+== What does ArgoCD do?
+
+You can use ArgoCD to provide declarative version control for application definitions, configurations, and environments in {product-title}.
+
+[id="argocd-support"]
+== Statement of support
+
+Use of ArgoCD with {product-title} is not supported.
```
Comment (kalexand-rh) [inline scalability_and_performance/argocd.adoc]: @aheslin, do we have standard language for these support statements?

File: _topic_map.yml [hunk 2]
```diff
@@ -1105,6 +1105,8 @@ Topics:
   File: routing-optimization
 - Name: What huge pages do and how they are consumed by apps
   File: what-huge-pages-do-and-how-they-are-consumed-by-apps
+- Name: Using ArgoCD with
```
Comment (vikram-redhat) [inline _topic_map.yml]: Missing {product-title}?

Comment (kalexand-rh) [reply _topic_map.yml]: That doesn't work in the topic map. :( I'll fix the typo, though.

Comment (vikram-redhat) [reply _topic_map.yml]: :) I typed that comment in without realizing which file it was in.

**Comment by kalexand-rh** [comment]
@vikram-redhat, @aheslin, @sferich888, @siamaksade, PTAL at the updated ArgoCD PR?

**Comment by siamaksade** [comment]
@kalexand-rh will we include a link to the github repo with the examples as well?
https://github.com/openshift/openshift-gitops-examples

**Comment by kalexand-rh** [comment]
> @kalexand-rh will we include a link to the github repo with the examples as well?
> https://github.com/openshift/openshift-gitops-examples

@siamaksade, that's a question for @sferich888. (IIRC, it was the content in GitHub _hard stop_ that Eric objected to.)

File: scalability_and_performance/argocd.adoc [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+ArgoCD enables you to deliver global custom resources, like the resources that are used to configure {product-title} clusters.
+
+[id="argocd-support"]
+== Statement of support
+
+Red Hat does not provide support for this tool. To obtain support for ArgoCD, see link:https://argoproj.github.io/argo-cd/SUPPORT/[Support] in the ArgoCD documentation.
+
+[id="argocd-documentation"]
+== ArgoCD documentation
```
Comment (siamaksade) [inline scalability_and_performance/argocd.adoc:22]: Could we add a section to call out that there is an operator for ArgoCD in OpenShift OperatorHub with links to the [operator installation docs](https://argocd--ocpdocs.netlify.app/openshift-enterprise/latest/operators/olm-adding-operators-to-cluster.html)?

**Comment by siamaksade** [comment]
> > @kalexand-rh will we include a link to the github repo with the examples as well?
> > https://github.com/openshift/openshift-gitops-examples
> 
> @siamaksade, that's a question for @sferich888. (IIRC, it was the content in GitHub _hard stop_ that Eric objected to.)

I thought we agreed to not presenting the content directly in openshift docs and just link to the repo so that it's clear the content in the GitHub repo is not supported.

File: scalability_and_performance/argocd.adoc (hunk 3 above)
Comment (kalexand-rh) [reply scalability_and_performance/argocd.adoc:22]: @aheslin, can we talk about this suggestion?

**Comment by siamaksade** [comment]
@kalexand-rh are there any blockers for adding ArgoCD repo in this doc and publishing it?

**Comment by kalexand-rh** [comment]
I checked with @aheslin, and there's no blocker to merging it as-is.

**Pull request merged**